### ISSUE TITLE
Always includes blank on select if use_required_attribute = true

### ISSUE THREAD

**Issue by diegocouto** [justinfrench/formtastic#997]
I can't remove the blank option from the select tags when use_required_attribute is being used. I tried to remove it locally:

``` ruby
<%= f.input :activity, as: :select, include_blank: false %>
```

and also changing the global configuration:

``` ruby
Formtastic::FormBuilder.include_blank_for_select_by_default = false
```

Unfortunately, none of these approaches worked as expected. The only whey to make it work is turning off the usage of the required attribute:

``` ruby
#Formtastic::FormBuilder.use_required_attribute = true
```

**Using**
- Rails 4.0.0
- Formtastic 2.2.1

**Comment by robbieshepherd** [comment]
@justinfrench I just hit this as well. 
`t.input :foo, as: :select, label: "Foo Type", include_blank: false, allow_blank: false, collection: {"Option 1" => "1", "Option 2" => "2"}, input_html: {required: true}` still renders a blank option in the select. Removing `required: true` makes the blank option then disappear.

**Comment by mikz** [comment]
@robbieshepherd could you try to contribute a failing test case?

**Comment by chewi** [comment]
I've figured it out. It took me a while as I started looking at this from the wrong end of the problem. This is deliberate behaviour in Rails. In fact, in Rails 5 (which I'm not using yet), this would actually raise an exception. See rails/rails@077540738460d44c1463b3e52d2acb1fed9bc766. They have taken this decision to comply with the HTML 5 spec. It does make some sense as the browser should guard you from clicking submit with an blank value selected but this doesn't help users on older browsers. I guess the only way to work around this is to leave the blank value in but select the first item in the collection by default.
